### ISSUE TITLE
Inflation does not work well using CG notation if the distribution has some unspecified (np.nan) probabilities

### ISSUE THREAD

**Issue by eliewolfe** [ecboghiu/inflation#130]
Suppose one is interested in the feasibility of a partially-specified distribution.
This would occur, for example, whenever two parties share the same setting: the distribution would only be known when the setting for the two parties coincides.
(This would also happen in cryptography applications, where an additional party Eve is assumed to be part of the observed distribution, but we only have observable statistics on some marginal. Though, in this example _no_ joint probabilities are known, only marginal probabilities are known.)
In this case using set_distribution with `include_all_outcomes=True` will give radically different results than with `include_all_outcomes=False`, as marginalization will fail when some of the joint probabilities are np.nan.

We should therefore **add a warning** that if np.nan is detected in set_distribution that the user should go back and re-initialize InflationLP or InflationSDP with `include_all_outcomes=True` .

**Comment by apozas** [comment]
If you need to do this, it can be already done using `set_values`, as in [Example 5](https://ecboghiu.github.io/inflation/_build/html/examples.html#Example-5:-Eavesdropped-quantum-repeater) in the documentation. Moreover, one can use the flags `use_lpi_constraints` and `only_specified_values` for choosing whether to just fix the values that are computed or also the products.

If you want to add a warning, go for it. But I guess that we will have to be careful when handling such distributions, so parts of `set_distribution` and `set_values` may need to be rewritten.

**Comment by eliewolfe** [comment]
I agree that it _can_ be done with `set_values`. That's why I am filing this an _enhancement_ more than _bug_.
I just think that `set_values` is awkward _relative_ to `set_distribution`, such that most users will want to use `set_distribution` whenever possible. And a distribution with some `np.nan`s totally _can_ be handled by `set_distribution` just as well as by `set_values` if --- and _only_ if --- `include_all_outcomes=True`.
No need to rewrite any code; we just need to add a warning. OK, I'll take a crack at it.
